detailed report, fix report creation

Arieg419 opened the pull request.

onlydole commented on gpt_researcher/orchestrator/agent/report_generator.py:
```diff
@@ -136,14 +138,19 @@ async def get_subtopics(self):
             )
 
-        subtopics_prompt = generate_subtopics_prompt().format(
+        # subtopics_prompt = generate_subtopics_prompt().format(
+        #     task=self.researcher.query,
+        #     data=self.researcher.context,
+        #     subtopics=self.researcher.subtopics,
+        #     max_subtopics=self.researcher.max_subtopics,
+        #     format_instructions="Return the subtopics as a list of strings."
+        # )
```
```suggestion
```

onlydole commented:
@Arieg419 - do you know if this passes the test suite (or if it runs via `docker compose`)?

Arieg419 commented:
hey @onlydole , i've moved this back to draft for now and am working through it --

The current state is that the 'detailed report' end report looks good but streams redundant intermediate steps to the UI. 

I tried to bisect and reverted to about two weeks ago, and I am also getting a janky, concise 'detailed report' (shorter than a basic report), so I am wondering when it last worked. If there was a known hash where it functioned well, I'm happy to work off that, @assafelovic; otherwise, I think I'll have a fix for this tomorrow.

Arieg419 commented:
cc @assafelovic detailed report looks good now, however, it's still missing references. will add tomorrow, and publish the PR for review.
